src/ape_geth/providers.py

```diff
@@ -139,7 +139,9 @@ class GethProvider(Web3Provider, UpstreamProvider):
 
     @property
     def uri(self) -> str:
-        return getattr(self.config, self.network.ecosystem.name)[self.network.name]["uri"]
+        ecosystem_config = getattr(self.config, self.network.ecosystem.name)
+        network_config = ecosystem_config.get(self.network.name, DEFAULT_SETTINGS)
+        return network_config.get("uri", DEFAULT_SETTINGS["uri"])
```

Comment:
Lol, using advice you gave in another PR against you:
```suggestion
        network_config = ecosystem_config.get(self.network.name) or DEFAULT_SETTINGS
        return network_config.get("uri") or DEFAULT_SETTINGS["uri"]
```